biblio-iacr.el

```diff
@@ -0,0 +1,91 @@
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+;;
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+;;
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+;;; Commentary:
+;;
+;; This file implements a backend for the for the `biblio' package (which see for more
+;; documentation).
+;;
+;;; Code:
+
+(require 'biblio-core)
+
+(defun biblio-iacr--forward-bibtex (metadata forward-to)
+  "Forward BibTeX for IACR entry METADATA to FORWARD-TO."
+    (funcall forward-to
+         (format "@misc{cryptoeprint:%s,\n    author = {%s},\n    title = {%s},\n    howpublished = {%s},\n    year = {%s},\n    note = {\\url{%s}},\n}"
+                 (biblio-alist-get 'eprintid metadata)
+                 (car (biblio-alist-get 'authors metadata))
+                 (biblio-alist-get 'title metadata)
+                 (biblio-alist-get 'container metadata)
+                 (biblio-alist-get 'year metadata)
+                 (biblio-alist-get 'url metadata))))
+
+(defun biblio-iacr--extract-interesting-fields (header body)
+  "Prepare a IACR search result, composed of HEADER and BODY, for display.
+HEADER contains the technical report id, which includes the year of publication.
+BODY includes the Title and Authors."
+  (let ((eprintid (caddr (nth 3 header))))
+        (list (cons 'year (substring eprintid 0 4))
+                (cons 'title (nth 2 (caddr body)))
+                (cons 'authors (list (nth 2 (nth 2 (nth 4 body)))))
+                (cons 'container (format "Cryptology ePrint Archive, Report %s" eprintid))
+                (cons 'type "eprint")
+                (cons 'references nil)
+                (cons 'eprintid eprintid)
+                (cons 'url (format "https://eprint.iacr.org/%s.pdf" eprintid)))))
+
+(defun biblio-iacr--traverse-results (list)
+  "Traverse the IACR html nodes LIST to produce a list of results recursively."
+         (if list
+                (cons (biblio-iacr--extract-interesting-fields (car list) (cadr list)) (biblio-iacr--traverse-results (cddr list)))
+                nil))
+
+(defun biblio-iacr--hitp (item tag)
+  "Check if ITEM from IACR response has html TAG."
+  (eq (car-safe item) tag))
+
+(defun biblio-iacr--parse-search-results ()
+  "Extract search results from IACR html response."
+  (biblio-decode-url-buffer 'utf-8)
+  (biblio-iacr--traverse-results (cddar (seq-filter (lambda (x) (biblio-iacr--hitp x 'dl))
+               (cdar (seq-filter (lambda (x) (biblio-iacr--hitp x 'body))
+                                 (libxml-parse-html-region (point-min) (point-max))))))))
+    ;; error warning ? h3: no reports found
```

Comment:
?

Reply:
This was a note to remind me of error handling which I forgot to take out, should be implemented now!